src/sumologic-span-processor/index.spec.ts

```diff
@@ -192,6 +195,33 @@ describe('SumoLogicSpanProcessor', () => {
       document.dispatchEvent(new Event('visibilitychange'));
       span.end(endTime); // end time is before page was hidden
       spanProcessor.onEnd(span);
+      jest.runAllTimers();
+
+      // Although page was hidden after span end time, we still marked it as 'hidden'.
+      // It's a special case that only applies for root spans.
+      // It was introduced to support `documentLoad` page visibility.
+      // Because `documentLoad` span ends when page was loaded, there are may be gaps in spans, so there would be no span
+      // where we may put the 'pagehide' event.
```

Comment:
Can you explain these gaps? Shouldn't the `documentLoad` span take the whole time from the beginning of loading the page until it's fully loaded?

Reply:
By OpenTelemetry `documentLoad` span ends when the `load` event is triggered and we don't want to break compatibility here. Because there are no attributes or events for traces, we want to save them in root spans.